sway/sway.5.scd

```diff
@@ -1058,6 +1075,22 @@ The following attributes may be matched with:
 	expression. If the value is \_\_focused\_\_, then all the views on the
 	currently focused workspace matches.
 
+*sandbox_engine*
+	Compare against the associated sandbox engine. Can be a regular expression.
+	If the value is \_\_focused\_\_, then the sandbox engine must be the same as
+	that of the currently focused window.
+
+*sandbox_app_id*
```

Comment:
I'm kind of wondering whether we should expose this, or whether we should always prefer the sandbox-provided app ID over the one provided by the sandboxed client.

(IIRC GNOME allows `org.libreoffice.Writer` if the sandbox app ID is `org.libreoffice`.)

Reply:
The sandbox `app_id` is defined to be engine-specific and may have an entirely different meaning.

For example, consider a system container (Docker, LXC, etc.) where the wayland socket / security context is associated with more than one application:
- `sandbox_engine` = "org.docker"
- `sandbox_app_id` = "my-docker-container-2"
- `app_id` = multiple, different per application

I'm not sure even that "Flatpak application ID" is equivalent to `app_id` (for that specific engine).

See https://github.com/swaywm/sway/pull/8521#issuecomment-2585655616

Reply:
That's a good point! Perhaps one could argue that any sandbox engine sending metadata through the security-context protocol is Wayland-aware (because it has to add support for that specific protocol)? Docker/LXC don't support Wayland apps out-of-the-box, so will not provide these app IDs which don't match xdg-shell. However, as you said nothing prevents a sandbox engine from having a completely different scheme for app IDs.

Reply:
My main point is that there exist use cases where a single wayland socket / security context must support multiple arbitrary applications, making it _impossible_ to set `sandbox_app_id` (a single value for the security context and all associated views) in a way which is compatible with `app_id` (potentially different per view).

For example, a system-level container (e.g. `docker run 'my-ubuntu-container-3'`) could start a terminal, browser, and media player via a single wayland socket / security context.

Flatpak (and other application-level container tools) are a special case where only one application is expected to be associated with a given security context (but again, I'm not sure that's completely accurate).

Reply:
System-level containers are less secure than per-app containers, because granted permissions apply to the whole container. That is, if a sandboxed app obtains a permission, all other apps in the container can abuse it.

But yeah, indeed, we don't want to prevent one to run containers with multiple apps in it.